src/llama-model.cpp

```diff
@@ -6681,8 +6668,9 @@ ggml_backend_buffer_type_t llama_model::select_buft(int il) const {
     return ::select_buft(
             *pimpl->dev_layer.at(il).buft_list,
             [&](ggml_context * ctx) {
-                ggml_tensor * cur = ggml_new_tensor_1d(ctx, GGML_TYPE_F32, hparams.n_embd);
-                ggml_tensor * layer_dir = ggml_new_tensor_1d(ctx, GGML_TYPE_F32, hparams.n_embd);
+                const int n_embd_inp = hparams.n_embd_inp();
+                ggml_tensor * cur = ggml_new_tensor_1d(ctx, GGML_TYPE_F32, n_embd_inp);
+                ggml_tensor * layer_dir = ggml_new_tensor_1d(ctx, GGML_TYPE_F32, n_embd_inp);
```

Comment:
I'm not 100% sure if `n_embd_inp` is correct here.

As far as I can understand, the `cur` is the output of the current layer. That means it should use the internal embedding length `n_embd`. CC @65a in case you know this part.